framework/src/executioners/Executioner.C

```diff
@@ -145,6 +149,8 @@ Executioner::Executioner(const InputParameters & parameters)
       getParam<bool>("compute_initial_residual_before_preset_bcs");
 
   _fe_problem.getNonlinearSystemBase()._l_abs_step_tol = getParam<Real>("l_abs_step_tol");
+
+  _fe_problem.setSNESMFReuseBase(getParam<bool>("snesmf_reuse_base"));
```

Comment:
You could add an overload of `FEProblemBase::setSNESMFReuseBase` that takes two arguments, with the second argument specifying whether the parameter was set by the user. So the method you would want to use is [`InputParameters::isParamSetByUser`](http://mooseframework.org/docs/doxygen/moose/InputParameters_8C_source.html#l00793). So here if you go with the overload, you could do:
```c++
_fe_problem.setSNESMFReuseBase(getParam<bool>("snesmf_reuse_base"), parameters.isParamSetByUser("snesmf_reuse_base"));
```

Reply:
I see.